src/software/ai/hl/stp/tactic/shadow_freekicker_tactic_test.cpp

```diff
@@ -34,6 +35,676 @@ TEST(ShadowFreekickerTacticTest, test_shadow_free_kicker_left_side)
 
     ASSERT_TRUE(action_ptr);
 
+    auto move_action = std::dynamic_pointer_cast<MoveAction>(action_ptr);
+    ASSERT_NE(nullptr, move_action);
+    // The edge of the robot should be slightly more than 0.5m from the ball
+    EXPECT_NEAR((world.ball().position() - move_action->getDestination()).length(), 0.62,
```

Comment:
could you clarify why we expect this to be around 0.62

Reply:
For sure, so this has some relation with our current implementation of the tactic. The `shadow freekicker tactic` would be assigned to two friendly robots to block the enemy's freekick. According to the rule, friendly robots cannot be closer than 0.5 m to the ball when an enemy is doing a freekick, so the distance between the ball with the line connecting two friendly robots is 0.5m with some room(1.05) and radius of the robot(0.09m) added to ensure that the edge of our robot won't be closer than 0.5 m to the ball during freekick.

I didn't find the reason why distance b is specified as 1.1 times of robot radius, but this is how `b` is calculated.

With `a` and `b` set up and known, the distance between the ball and friendly robot `c` then become 0.62292, which is close to 0.62.

![image](https://user-images.githubusercontent.com/43716969/76593714-4f90c500-64b4-11ea-87d1-7099a1154961.png)

Reply:
We should at least add a comment saying essentially this, just so that anyone looking at the code in the future can figure it out.

Reply:
this information is added as an ascii graph